test/sequential/feeLock.API.feePayment.test.ts

```diff
@@ -0,0 +1,399 @@
+      checkMgaWhitelistedToken = true;
+    }
+  });
+
+  if (checkMgaWhitelistedToken) {
+    const checkEmptyTimeoutConfig = await updateFeeLockMetadata(
+      sudo,
+      new BN(periodLength),
+      new BN(feeLockAmount),
+      swapValueThreshold,
+      [[MGA_ASSET_ID, false]]
+    );
+    await waitSudoOperataionSuccess(checkEmptyTimeoutConfig);
+  }
+}
+
+async function addMgaToWhitelisted() {
+  let checkMgaWhitelistedToken: boolean;
+  const swapValueThreshold = feeLockMetadata.swapValueThreshold;
+  whitelistedTokens = feeLockMetadata.whitelistedTokens;
+  checkMgaWhitelistedToken = false;
+
+  whitelistedTokens.forEach((element) => {
+    if (element.toString() === MGA_ASSET_ID.toString()) {
+      checkMgaWhitelistedToken = true;
+    }
+  });
+
+  if (swapValueThreshold < thresholdValue) {
+    checkMgaWhitelistedToken = false;
+  }
+
+  if (!checkMgaWhitelistedToken) {
+    const updateMgaTimeoutMetadata = await updateFeeLockMetadata(
+      sudo,
+      new BN(periodLength),
+      new BN(feeLockAmount),
+      thresholdValue,
+      [[MGA_ASSET_ID, true]]
+    );
+    await waitSudoOperataionSuccess(updateMgaTimeoutMetadata);
+  }
+}
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  await sudo.addMGATokens(sudo);
+
+  firstCurrency = await Assets.issueAssetToUser(
+    sudo,
+    defaultCurrencyValue,
+    sudo
+  );
+});
+
+beforeEach(async () => {
+  const api = getApi();
+
+  [testUser1] = setupUsers();
+
+  await setupApi();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(firstCurrency, testUser1, defaultCurrencyValue),
+    Assets.mintNative(sudo),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        defaultPoolVolumeValue,
+        firstCurrency,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+
+  feeLockMetadata = JSON.parse(
+    JSON.stringify(await api.query.feeLock.feeLockMetadata())
+  );
+
+  periodLength = feeLockMetadata.periodLength;
+  feeLockAmount = feeLockMetadata.feeLockAmount;
+
+  if (periodLength > 20) {
+    await updateFeeLockMetadata(
+      sudo,
+      new BN(10),
+      new BN(feeLockAmount),
+      null,
+      null
+    );
+
+    feeLockMetadata = JSON.parse(
+      JSON.stringify(await api.query.feeLock.feeLockMetadata())
+    );
+
+    periodLength = feeLockMetadata.periodLength;
+  }
+});
+
+test("gassless- GIVEN a feeLock configured (only Time and Amount ) WHEN the user swaps AND the user has not enough MGAs and has enough TURs THEN the extrinsic fails on submission", async () => {
+  await clearMgaFromWhitelisted();
+
+  await testUser1.addTURTokens(sudo);
+  await testUser1.addMGATokens(sudo, new BN(2));
+
+  await checkErrorSellAsset(
+    "1010: Invalid Transaction: Fee lock processing has failed either due to not enough funds to reserve or an unexpected error"
```

Comment:
Would you mind creatign a costant with this error message, looks recurrent and can be shared with all the tests.